src/agent.py

```diff
@@ -576,13 +576,30 @@ def get_batch_gp_sensitivities(self, xu_hat, sqp_iter):
         for i in range(g_xu_hat.shape[1] - 1):
             assert torch.all(g_xu_hat[:, i + 1, :, :] == g_xu_hat[:, i, :, :])
 
-        if self.params["agent"]["true_dyn_as_sample"] and self.params["agent"]["num_dyn_samples"] == 1:
+        update_hallucinated_data = True
+        if (self.params["agent"]["true_dyn_as_sample"] or self.params["agent"]["mean_as_dyn_sample"]) and self.params["agent"]["num_dyn_samples"] == 1:
             y_sample = torch.zeros((1, self.g_ny, self.params["optimizer"]["H"], self.in_dim_y))
+            update_hallucinated_data = False
+        elif (self.params["agent"]["true_dyn_as_sample"] and self.params["agent"]["mean_as_dyn_sample"]) and self.params["agent"]["num_dyn_samples"] == 2:
+            y_sample = torch.zeros((2, self.g_ny, self.params["optimizer"]["H"], self.in_dim_y))
+            update_hallucinated_data = False
         else:
             y_sample = self.sample_gp(
                 g_xu_hat, base_samples=self.epistimic_random_vector[self.mpc_iter][sqp_iter]
             )
             # y_sample = self.sample_gp(g_xu_hat)
+        if not update_hallucinated_data:
```

Comment:
A quick question: 
If we have only a true dynamics sample; Do we still need to call the model (enter in here) for something else?

Reply:
Hmm good point. This is only necessary when the mean is to be evaluated, so technically the condition "and self.params["agent"]["mean_as_dyn_sample"])" could be added. Does this make sense?

Reply:
okay sounds good, Thanks. I can make the change.